Functions chapter 12.playground/Contents.swift

```diff
@@ -9,6 +9,86 @@ func printgreeting () {
 }
 printgreeting()
 
-// Function Parameters
+// Function Parameters 
+//Declare  function called programming, give it no parameters, but in the {} brackets make it print something to the screen about programming. Then, Call this function, remember just start by typing the function name, outside of this function :    and to end this  To help you understand the flow of a program, put a print statement before the function call, one at the beginning of the function, and one after the function call...you will now see how calling a function changes the flow (hopefully)
+
+
+
+func programming1 () {
+    print (" I program all night")
+    
+}
+
+print ("I program all night")
+programming1()
+print ("I program all night")
+
+//Now give this function a parameter (tiredOfProgramming), make the parameter of BOOL type.
+//Notice your function call from (3) will no longer work, so delete the function call, and start typing programming, you will notice how XCode auto completes the function for you, press return, just remember to fill in a value for the parameter        
+
+func programming2 (tiredOfProgramming: Bool) {
+    print("Then I program all night")
+    
+}
+
+programming2(tiredOfProgramming: true)
+
+//Now check the value of this parameter in the function using and IF statement on the Bool parameter tiredOfProgramming, you can change the text you print if you like! 
+
+func programming3 (tiredOfProgramming: Bool) -> String {
+    
+    guard tiredOfProgramming  else {
+        return "Ride a meditation cushion, it helps relax"
+    }
+    return "I program all night"
+}
+    
+
+
+print (programming3(tiredOfProgramming: true))
+
+//Play around with changing the parameter you pass in, to either true or false, see what different print statements occur
+
+
+//....Comments: when parameters change the answer as well or what is printed is different if it is "false" we Then i code all night and if it "true" i ride the meditation cushion.
+
+//Now we want to return the String instead of printing it out in a function...add the syntax on the function declaration that allows return of type String:
+
+//Can you see why this is neater? It returns immediately, when your functions get very big and complex this makes more sense as it prevents the whole function being carried out.
+
+func programming4 (tiredOfProgramming tired: Bool) -> String {
+    guard tired else {
+        return "ride a meditation cushion, it helps relax"
+    }
+        return "I program all night"
+}
```

Comment:
last return statement should me indented one tab to the left as it is outside the guard brackets {} e.g.


```
func programming4 (tiredOfProgramming tired: Bool) -> String {
     guard tired else {
         return "ride a meditation cushion, it helps relax"
     }
     return "I program all night"
 }
```